dask/dataframe/tests/test_dataframe.py

```diff
@@ -812,6 +812,17 @@ def test_map_partitions_type():
     assert all(x == pd.DataFrame for x in result)
 
 
+def test_map_partitions_partition_info():
+    def f(df, partition_info=None):
+        assert partition_info is not None
+        assert "number" in partition_info
```

Comment:
It'd be good to check that the number and division have the correct values as well.